test/cmd/command_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+func TestRunCommand(t *testing.T) {
+	testCases := []struct {
+		command string
+		expectedOutput string
+		expectedErrorCode int
+	}{
+		{"", "", 1},
+		{"   ", "", 1},
+		{"echo hello, world", "hello, world\n", 0},
+		{"unknowncommand", "", 1},
```

Comment:
This looks fishy. I would expect 127 for an unknown binary.

Reply:
The `os/exec` library does not return `exec.ExitError` for unknown commands, so we return 1 as the default error code.

This test case is not that useful as it does not cover the `exec.ExitError`, changed to the command like `bash -c 'echo foo > /dev/stderr; exit 4'`